From 0b9db98aea32f5d634de8268a57b442f41fffc45 Mon Sep 17 00:00:00 2001
From: Jiawei Huang <jiawei@tigera.io>
Date: Wed, 15 Jul 2026 11:10:33 +0800
Subject: [PATCH] windows: fix MinGW-w64 build of win_file.c

Backport of bytecodealliance/wasm-micro-runtime@0b9db98aea32f5d634de8268a57b442f41fffc45.
---
 core/shared/platform/windows/win_file.c | 35 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 18 deletions(-)

diff --git a/core/shared/platform/windows/win_file.c b/core/shared/platform/windows/win_file.c
index 34a1260b..8626351b 100644
--- a/core/shared/platform/windows/win_file.c
+++ b/core/shared/platform/windows/win_file.c
@@ -7,7 +7,7 @@
 #include "libc_errno.h"
 #include "win_util.h"
 
-#include "PathCch.h"
+#include "pathcch.h"
 
 #pragma comment(lib, "Pathcch.lib")
 
@@ -1295,26 +1295,25 @@ os_readlinkat(os_file_handle handle, const char *path, char *buf,
 
         if (wbufsize >= 4 && wbuf[0] == L'\\' && wbuf[1] == L'?'
             && wbuf[2] == L'?' && wbuf[3] == L'\\') {
-            // Starts with \??\ 
+            // Starts with \??\ prefix
             if (wbufsize >= 6
                 && ((wbuf[4] >= L'A' && wbuf[4] <= L'Z')
                     || (wbuf[4] >= L'a' && wbuf[4] <= L'z'))
-                && wbuf[5] == L':' && (wbufsize == 6 || wbuf[6] == L'\\'))
-                {
-                    // \??\<drive>:\ 
-                    wbuf += 4;
-                    wbufsize -= 4;
-                }
-                else if (wbufsize >= 8 && (wbuf[4] == L'U' || wbuf[4] == L'u')
-                         && (wbuf[5] == L'N' || wbuf[5] == L'n')
-                         && (wbuf[6] == L'C' || wbuf[6] == L'c')
-                         && wbuf[7] == L'\\')
-                {
-                    // \??\UNC\<server>\<share>\ - make sure the final path looks like \\<server>\<share>\ 
-                    wbuf += 6;
-                    wbuf[0] = L'\\';
-                    wbufsize -= 6;
-                }
+                && wbuf[5] == L':' && (wbufsize == 6 || wbuf[6] == L'\\')) {
+                // \??\<drive>:\ form
+                wbuf += 4;
+                wbufsize -= 4;
+            }
+            else if (wbufsize >= 8 && (wbuf[4] == L'U' || wbuf[4] == L'u')
+                     && (wbuf[5] == L'N' || wbuf[5] == L'n')
+                     && (wbuf[6] == L'C' || wbuf[6] == L'c')
+                     && wbuf[7] == L'\\') {
+                // \??\UNC\<server>\<share>\ - make sure the final path looks
+                // like \\<server>\<share>\ afterwards
+                wbuf += 6;
+                wbuf[0] = L'\\';
+                wbufsize -= 6;
+            }
         }
     }
     else if (reparse_data->ReparseTag == IO_REPARSE_TAG_MOUNT_POINT) {
-- 
2.50.1

